common/src/test/java/io/netty/util/concurrent/UnorderedThreadPoolEventExecutorTest.java

```diff
@@ -54,4 +55,17 @@ public void operationComplete(Future<Object> future) throws Exception {
             executor.shutdownGracefully();
         }
     }
+
+    @Test(timeout = 10000)
+    public void scheduledAtFixedRateMustRunTaskRepeatedly() throws InterruptedException {
+        UnorderedThreadPoolEventExecutor executor = new UnorderedThreadPoolEventExecutor(1);
+        final CountDownLatch latch = new CountDownLatch(3);
+        executor.scheduleAtFixedRate(new Runnable() {
+            @Override
+            public void run() {
+                latch.countDown();
+            }
+        }, 1, 1, TimeUnit.MILLISECONDS);
+        latch.await();
```

Comment:
After this unblocks we should cancel the scheduled task via `future.cancel()`. Also we should shutdown the `executor`.